### PR TITLE
everything is migrated to tsoa

### PR DESCRIPTION
**Pull request by chitalian**

### CODE REVIEW

File: valhalla/jawn/src/controllers/datasetController.ts [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+  if (!org.tier) {
+    return false;
+  }
+  if (org.tier === "free") {
+    const jobCountQuery = await supabaseServer.client
+      .from("finetune_job")
+      .select("*", { count: "exact" })
+      .eq("organization_id", orgId);
+    console.log("jobCountQuery", jobCountQuery);
```
Comment (colegottdank) [inline valhalla/jawn/src/controllers/datasetController.ts:44]: delete

File: valhalla/jawn/src/controllers/datasetController.ts [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    const jobsCount = jobCountQuery.count ?? 1;
+    console.log("jobsCount", jobsCount);
```
Comment (colegottdank) [inline valhalla/jawn/src/controllers/datasetController.ts:46]: delete

File: valhalla/jawn/src/controllers/datasetController.ts [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+@Security("api_key")
+export class DatasetController extends Controller {
+  @Post("{datasetId}/fine-tune")
+  @Tags("FineTune")
+  public async datasetFineTune(
+    @Path() datasetId: string,
+    @Body()
+    body: FineTuneBodyParams,
+    @Request() request: JawnAuthenticatedRequest
+  ): Promise<
```
Comment (colegottdank) [inline valhalla/jawn/src/controllers/datasetController.ts:70]: Would be cool to extract this response body of error, success, data

File: valhalla/jawn/src/controllers/datasetController.ts [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+    let filterNode: FilterNode;
+    try {
+      filterNode = JSON.parse(dataset?.filter_node ?? "");
+    } catch (e) {
+      this.setStatus(500);
+      return {
+        error: "No dataset found",
+      };
+    }
+    if (datasetError || !dataset) {
```
Comment (colegottdank) [inline valhalla/jawn/src/controllers/datasetController.ts:105]: we could just check this prior to the try catch above

File: valhalla/jawn/src/lib/clients/postHogClient.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+const ph_project_api_key = process.env.PUBLIC_POSTHOG_API_KEY;
+export let postHogClient: PostHog | null = null;
+if (ph_project_api_key) {
+  postHogClient = new PostHog(ph_project_api_key, {
+    host: "https://app.posthog.com",
+  });
+}
+
+process.on("exit", () => {
```
Comment (colegottdank) [inline valhalla/jawn/src/lib/clients/postHogClient.ts:11]: dope

**Comment by colegottdank** [comment]
OTHERWISE LGTM

File: valhalla/jawn/src/controllers/datasetController.ts (hunk 1 above)
Comment (chitalian) [reply valhalla/jawn/src/controllers/datasetController.ts:44]: ugh not your commented on migrated code  lol . jk it's all good

File: valhalla/jawn/src/lib/clients/postHogClient.ts (hunk 5 above)
Comment (chitalian) [reply valhalla/jawn/src/lib/clients/postHogClient.ts:11]: thank you

**Pull request merged**